### PR TITLE
deploy-31 Implement DB changes for Umbraco Deploy

### PR DESCRIPTION
**Pull request by Shazwazza**

### CODE REVIEW

File: src/Umbraco.Core/Models/Rdbms/UmbracoDeployChecksumDto.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+        [Column("entityPath")]
+        [Length(256)]
+        [NullSetting(NullSetting = NullSettings.Null)]
+        public string EntityPath { get; set; }
+
+        [Column("localChecksum")]
+        [NullSetting(NullSetting = NullSettings.NotNull)]
+        [Length(32)]
+        public string LocalChecksum { get; set; }
```
Comment (zpqrtbnk) [inline src/Umbraco.Core/Models/Rdbms/UmbracoDeployChecksumDto.cs]: I think the path should be 512 - and the checksum could be 128.

File: src/Umbraco.Core/Models/Rdbms/UmbracoDeployDependencyDto.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+namespace Umbraco.Core.Models.Rdbms
+{
+    [TableName("umbracoDeployDependency")]
+    [PrimaryKey("id")]
+    [ExplicitColumns]
+    internal class UmbracoDeployDependencyDto
+    {
+        [Column("id")]
+        [PrimaryKeyColumn(Name = "PK_umbracoDeployDependency")]
+        public int Id { get; set; }
```
Comment (zpqrtbnk) [inline src/Umbraco.Core/Models/Rdbms/UmbracoDeployDependencyDto.cs]: Can't we really have a composite PK on sourceId, targetId?

File: src/Umbraco.Core/Models/Rdbms/UmbracoDeployChecksumDto.cs (hunk 1 above)
Comment (Shazwazza) [inline src/Umbraco.Core/Models/Rdbms/UmbracoDeployChecksumDto.cs]: but a 128 bit hash = 32 string char. a 128 length checksum would be a massive bit hash

File: src/Umbraco.Core/Models/Rdbms/UmbracoDeployDependencyDto.cs (hunk 2 above)
Comment (Shazwazza) [inline src/Umbraco.Core/Models/Rdbms/UmbracoDeployDependencyDto.cs]: yeah, we can do that, just wasn't sure if those were supposed to be the unique keys in this situation.

File: src/Umbraco.Core/Models/Rdbms/UmbracoDeployChecksumDto.cs (hunk 1 above)
Comment (zpqrtbnk) [inline src/Umbraco.Core/Models/Rdbms/UmbracoDeployChecksumDto.cs]: My mistake, current Deploy uses 128bits, 32 chars indeed.

**Pull request merged**